ote/src/cljs/ote/views/rental.cljs

```diff
@@ -218,90 +232,86 @@
     :rows 1}))
 
 (defn service-hours-for-location [update-form! data]
-  (reagent/with-let [open? (reagent/atom false)]
-    [:div
-     [ui/flat-button {:label (tr [:rentals-page :open-service-hours-and-exceptions])
-                      :on-click #(reset! open? true)}]
-     [ui/dialog
-      {:open @open?
-       :actionsContainerStyle style-dialog/dialog-action-container
-       :auto-scroll-body-content true
-       :title (tr [:opening-hours-dialog :header-dialog])
-       :actions [(reagent/as-element
-                  [ui/flat-button {:label (tr [:buttons :close])
-                                   :on-click #(reset! open? false)}])]}
-      [form/form {:update! update-form!
-                  :name->label (tr-key [:field-labels :rentals]
-                                       [:field-labels :transport-service]
-                                       [:field-labels])}
-       [(assoc-in (ts-common/service-hours-group "rental") [:options :card?] false)]
-       data]]
-     ]))
+  [form/form {:update! update-form!
+              :name->label (tr-key [:field-labels :rentals]
+                                   [:field-labels :transport-service]
+                                   [:field-labels])}
+   [(ts-common/service-hours-group "rental")]
+   data])
+
+(defn service-hour-form-element
+  "Define immutable function for service hours (and exceptions) so that reagent won't re-render the whole component in every
```

Comment:
Would setting a custom ^{:key ...} have worked as an alternative?

Reply:
Nope. This function is in our form component which is rendered via Reagent and using key wouldn't help at all. I tested it. It did run the function again every time and the same key wouldn't help.